aws-curl

```diff
@@ -499,6 +507,22 @@ fi
 if [ "$EC2_CREDS" = 1 ]; then
     ec2_import_creds
 fi
+get_cred_value() {
+    echo "$1" | grep "$2 =" | cut -d ' ' -f 3-
+}
+
+if [ -n "$AWS_PROFILE" ] && [ -z "$AWS_ACCESS_KEY_ID" ]; then
+  # this can work with AWS SSO based connections
+  block=$(sed -n '/\['$AWS_PROFILE'/,/^$/p' ~/.aws/credentials)
```

Comment:
Mac needs `gsed`, there is SED_CMD env variable that is populated on mac with gsed that you can use instead of referencing to `sed`.

There is missing closing `\]`.

Selecting lines from section start to first empty line likely is not like aws cli parses this file. Likely section end condition is one of 1) start of new section 2) end of file. This regexp need a fix.